### PR TITLE
PR provider refactor new comments

### PR DESCRIPTION
**Pull request by Buhana149**

### CODE REVIEW

File: lib/ui_utilities/city_selection_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_typeahead/flutter_typeahead.dart';
+
+class CitySelectionDialog extends StatelessWidget {
+  final Widget Function(BuildContext, Object?) itemBuilder;
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/city_selection_dialog.dart:7]: Missing new line 
class CitySelectionDialog extends StatelessWidget {

  final Widget Function(BuildContext, Object?) itemBuilder;

File: lib/ui_utilities/city_selection_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  final void Function(Object?)? onSelected;
+  final FutureOr<List<Object?>?> Function(String) suggestionsCallback;
+  final void Function()? onPressedCancel;
+  final void Function()? onPressedSubmit;
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/city_selection_dialog.dart:11]: Missing new line
final void Function()? onPressedSubmit;

  const CitySelectionDialog({

File: lib/ui_utilities/city_selection_dialog.dart (hunk 1 above)
Comment (Buhana149) [reply lib/ui_utilities/city_selection_dialog.dart:7]: done

File: lib/ui_utilities/city_selection_dialog.dart (hunk 2 above)
Comment (Buhana149) [reply lib/ui_utilities/city_selection_dialog.dart:11]: done

File: lib/pages/home_page.dart [hunk 3]
```diff
@@ -31,6 +33,34 @@ class _HomePageState extends State<HomePage> {
   @override
   Widget build(BuildContext context) {
     return Consumer<WeatherProvider>(builder: (context, value, child) {
+      void showCitySelectionDialog() {
```
Comment (vlad-buhaescu) [inline lib/pages/home_page.dart]: ❗❗❗ let's move this method outside of build

**Pull request merged**